gossip/mempool.go

```diff
@@ -0,0 +1,25 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package gossip
+
+// Tx is a transaction that can be gossiped across the network.
+type Tx interface {
+	// Hash returns the unique hash of this transaction
+	Hash() []byte
```

Comment:
should we make this an `ids.ID` type for clarity and to guarantee that it's a specific length? I think this would be a reasonable constraint on the interface and we already assume it's at least 8 bytes in the hasher

Reply:
For some reason I thought that geth's tx hash lengths and our hash lengths were different so I used `[]byte` but it looks like they're both 32 bytes. Should be able to do this 👍 .